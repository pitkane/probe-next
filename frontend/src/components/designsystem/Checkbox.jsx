import React from "react";
import PropTypes from "prop-types";

const Checkbox = props => {
  const { value, id } = props;

  return (
    <div className="uk-margin">
      <label htmlFor={id}>
        <input id={id} className="uk-checkbox" type="checkbox" /> {value}
      </label>
    </div>
  );
};

Checkbox.propTypes = {
  value: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired
};

Checkbox.defaultProps = {};

export default Checkbox;
