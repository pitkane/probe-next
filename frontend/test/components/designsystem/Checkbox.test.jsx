import React from "react";
import renderer from "react-test-renderer";

import Checkbox from "../../../src/components/designsystem/Checkbox";

describe("Checkbox", () => {
  it("renders correctly", () => {
    const component = <Checkbox />;
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
