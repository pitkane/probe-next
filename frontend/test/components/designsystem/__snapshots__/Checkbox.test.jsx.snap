// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders correctly 1`] = `
<div
  className="uk-margin"
>
  <label
    htmlFor=""
  >
    <input
      className="uk-checkbox"
      id=""
      type="checkbox"
    />
     
    
  </label>
</div>
`;
